core/src/test/scala/org/apache/spark/FileSuite.scala

```diff
@@ -700,4 +701,38 @@ class FileSuite extends SparkFunSuite with LocalSparkContext {
 
     assert(collectRDDAndDeleteFileBeforeCompute(true).isEmpty)
   }
+
+  test("SPARK-25100: Using KryoSerializer and" +
+      "setting registrationRequired true can lead job failed") {
+    val tempDir = Utils.createTempDir()
```

Comment:
There's `tempDir` which FileSuite handles the directory creation per test - let's rely on it.